jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamSchemasResultSet.java

```diff
@@ -15,22 +15,100 @@
+  private static final Datum NULL_DATUM = new Datum().withNullValue(Boolean.TRUE);
   private static final List<ColumnInfo> COLUMNS = ImmutableList.of(
     TimestreamDataType.createColumnInfo(TimestreamDataType.VARCHAR, "TABLE_SCHEM"),
     TimestreamDataType.createColumnInfo(TimestreamDataType.VARCHAR, "TABLE_CATALOG"));
 
+  private boolean isAfterLast = false;
+
   /**
    * Constructor.
+   *
+   * @param connection the parent connection of the result set.
+   * @param schemaPattern the SchemaPattern. null if not provided
+   * @throws SQLException if a database access error occurs.
    */
-  public TimestreamSchemasResultSet() {
-    super(COLUMNS);
+  TimestreamSchemasResultSet(TimestreamConnection connection, String schemaPattern) throws SQLException {
+    super(null, 20);
+    this.rsMeta = createColumnMetadata(COLUMNS);
+
+    populateCurrentRows(connection, schemaPattern);
+  }
+
+  @Override
+  public boolean isAfterLast() throws SQLException {
+    verifyOpen();
+    return isAfterLast;
+  }
+
+  @Override
+  public boolean isLast() throws SQLException {
+    verifyOpen();
+    LOGGER.debug("Checking whether the last row of this TimestreamSchemasResultSet has been reached.");
+    return !isAfterLast && !this.rowItr.hasNext();
+  }
+
+  @Override
+  protected void doClose() {
+    LOGGER.debug("Closed is called on this TimestreamSchemasResultSet, do nothing as the result set has already been closed.");
+  }
+
+  /**
+   * Retrieve the next page of the results.
+   *
+   * @return {@code true} if there is another page; {@code false} otherwise.
+   */
+  @Override
+  protected boolean doNextPage() {
+    LOGGER.debug("Attempting to retrieve the next page of results. There are no more pages, return false.");
+    this.isAfterLast = true;
+    return false;
+  }
+
+  /**
+   * Map the list of databases into a Timestream Row type to allow reuse of the common ResultSet
+   * retrieval path.
+   *
+   * @param connection The parent connection to retrieve databases from.
+   * @param schemaPattern The schemaPattern to filter databases
+   * @throws SQLException if there is an error listing the databases.
+   */
+  private void populateCurrentRows(TimestreamConnection connection, String schemaPattern) throws SQLException {
+    final List<Row> databases = new ArrayList<>();
+    try (Statement statement = connection.createStatement()) {
+      LOGGER.debug("Retrieving a list of databases." + (Strings.isNullOrEmpty(schemaPattern) ? "" : " Schema pattern is " + schemaPattern + "."));
+      final String query = "SHOW DATABASES" +
+              (Strings.isNullOrEmpty(schemaPattern) ? "" : " LIKE '" + schemaPattern + "'");
+      try (ResultSet rs = statement.executeQuery(query)) {
+        while (rs.next()) {
+          databases.add(new Row().withData(
+                  new Datum().withScalarValue(rs.getString(1)),
+                  NULL_DATUM
+          ));
```

Comment:
why NULL_DATUM here? This is a behavior change from getCatalogs()?

Reply:
I believe, it is for the catalog that gets returned as `null`? just ensuring.

Reply:
Yes, it is for catalog.